Source/core/JSONRPC.h

```diff
@@ -260,14 +260,19 @@ namespace Core {
                 size_t end = designator.find_last_of('@');
                 size_t begin = designator.find_last_of('.', end);
                 size_t lookup = designator.find_first_of('#', begin+1);
+                string method;
 
                 if (lookup != string::npos) {
                     size_t ns = designator.find_first_of(':', lookup + 1);
-                    return (designator.substr((begin == string::npos) ? 0 : begin + 1, lookup - begin - 1) + (ns != string::npos? designator.substr(ns, end - ns) : string{}));
+                    method = designator.substr((begin == string::npos) ? 0 : begin + 1, lookup - begin - 1) + (ns != string::npos? designator.substr(ns, end - ns) : string{});
                  }
                 else {
-                    return (designator.substr((begin == string::npos) ? 0 : begin + 1, (end == string::npos ? string::npos : (begin == string::npos) ? end : end - begin - 1)));
+                    method = designator.substr((begin == string::npos) ? 0 : begin + 1, (end == string::npos ? string::npos : (begin == string::npos) ? end : end - begin - 1));
                 }
+
```

Comment:
@sebaszm maybe add a DEPRECATED in the comment (just above this method ToCamelCase) so in R6 we can search for the DEPRECATED keywords and know this might require dropping this feature..